state/executor.go

```diff
@@ -589,7 +603,48 @@ func (t *Transition) applyCall(
 		}
 	}
 
-	result := t.run(c, host)
+	var result *runtime.ExecutionResult
+
+	if t.ctx.Tracer != nil {
+		// XXX: Depth begins from 0 in geth
+		if c.Depth == 1 {
```

Comment:
Not sure if I follow this. What is the difference between `CallStart` and `InnerCallStart`? Can we merge both? The `Tracer` implementation for `InnerCallStart` is void. Does this mean that only calls at depth 1 would be tracked?

Reply:
Merged 9654a544c54118078600a0b83bddc5ab3c1a24d5